File: tlmanager/test/insert_many_data.py
import time
import requests

def insert_leaf(tree_name, data):
    URL = 'http://localhost:8000/insert-leaf'
    data = {
        'tree-name': tree_name,
        'data': data
    }
    response = requests.post(URL, json=data)
    return response

def create_tree(tree_name, commitment_size):
    URL = 'http://localhost:8000/tree-create'
    data = {
        'tree-name': tree_name,
        'commitment-size': commitment_size
    }
    response = requests.post(URL, json=data)
    return response

def commit_tree(tree_name):
    URL = 'http://localhost:8000/tree/commit'
    data = {
        'tree-name': tree_name
    }
    response = requests.post(URL, json=data)
    return response

if __name__ == '__main__':
    start = time.time()
    create_tree('tree1', 2048)
    for i in range(1000000):
        print(i, end='\r')
        response = insert_leaf('tree1', str(i))
        if response.status_code != 200:
            print(response.json())
            break
    commit_tree('tree1')
    end = time.time()
    print(f'Time elapsed: {end - start}')

File: tlmanager/test/js/insert_many_data.js
const axios = require('axios');

const createTree = async (tree_name, commitment_size) => {
    return axios.post('http://localhost:8000/tree-create', {
        "tree_name": tree_name,
        "commitment_size": commitment_size
    }).then((res) => {
        console.log(res.data);
    }).catch((err) => {
        console.log(err);
    })
}

const insertLeaf = (tree_name, data) => {
    return axios.post('http://localhost:8000/insert-leaf', {
        "tree_name": tree_name,
        "data": data
    }).then((res) => {
        //console.log(res.data);
    }).catch((err) => {
        //console.log(err);
    })
}

const commitTree = (tree_name) => {
    return axios.post('http://localhost:8000/tree/commit', {
        "tree_name": tree_name
    })
    .then((res) => {
        console.log(res.data);
    }).catch((err) => {
        console.log(err);
    })
    
}

const main = async () => {
    const start = new Date().getTime();
    await createTree("tree1", 2048);
    const inserts = [];
    for (let i = 0; i < 1000; i++) {
        for (let j = 0; j < 1000; j++) {
            inserts.push(insertLeaf("tree1", "data" + i + j));
        }
        await Promise.all(inserts);
        inserts.length = 0;
    }
    await commitTree("tree1");
    const end = new Date().getTime();
    console.log("Time taken: ", end - start, "ms"); 
}

main();